index.js

```diff
@@ -37,7 +37,14 @@ const withCompileNodeModules = (userOptions = {}) => (nextConfig = {}) => ({
         const jsRuleNodeModules = copyJsRule(jsRule, userOptions);
 
         config.module.rules.splice(jsRuleIndex + 1, 0, jsRuleNodeModules);
-        delete config.externals;
+
+        if (options.isServer) {
+            // This is needed since Next.js requires the React to be the same instance in every page.
+            // Otherwise, React would be injected individually in every page and using
+            // React Hooks would throw: Invalid Hook Call Warning (https://reactjs.org/warnings/invalid-hook-call-warning.html)
+            // Regex copied from https://github.com/zeit/next.js/blob/canary/packages/next/build/webpack-config.ts#L295
```

Comment:
I would change this to a permalink otherwise the same line might not have that regexp anymore.

Reply:
You're right! Done ✅